File: 백준/Bronze/25501. 재귀의 귀재/재귀의 귀재.py
import sys


def recursion(s, l, r):
    global cnt
    cnt += 1
    if l >= r:
        return 1
    elif s[l] != s[r]:
        return 0
    else:
        return recursion(s, l + 1, r - 1)


def is_palindrome(s):
    return recursion(s, 0, len(s) - 1)


t = int(sys.stdin.readline())
for _ in range(t):
    s = sys.stdin.readline().rstrip()
    cnt = 0
    print(is_palindrome(s), cnt)

File: 백준/Bronze/25501. 재귀의 귀재/재귀의 귀재.cc
#include <iostream>
#include <cstring>
using namespace std;

int cnt = 0;

int recursion(const char *s, int l, int r) {
	cnt++;
	if (l >= r) return 1;
	else if (s[l] != s[r]) return 0;
	return recursion(s, l + 1, r - 1);
}

int isPalindrome(const char *s) {
	return recursion(s, 0, strlen(s) - 1);
}

int main() {
	cin.tie(0)->sync_with_stdio(0);

	int t;
	char s[1'001];
	cin >> t;
	while (t--) {
		cnt = 0;
		cin >> s;
		cout << isPalindrome(s);
		cout << ' ' << cnt << '\n';
	}
}